packages/flutter_tools/lib/src/artifacts.dart

```diff
@@ -437,6 +439,10 @@ class CachedArtifacts implements Artifacts {
       case TargetPlatform.linux_arm64:
       case TargetPlatform.windows_x64:
         return _getDesktopArtifactPath(artifact, platform, mode);
+      // Artifacts for windows-arm64 are not yet available.
+      // TODO(stuartmorgan): https://github.com/flutter/flutter/issues/62597
+      case TargetPlatform.windows_arm64:
+        return _getDesktopArtifactPath(artifact, TargetPlatform.windows_x64, mode);
```

Comment:
I'm confused; why are you returning x64 artifacts when targeting ARM? If there aren't ARM artifacts yet, then this shouldn't land. The order should be to update the engine repo to build and upload the necessary artifacts, then upload the tool to use them.

Reply:
For now, on arm64, it should behave like x64 (because artifacts are not available). As mentioned in the commit message, you need a prebuilt arm64 flutter engine (built by hand) to create an arm64 app.

I'd love to have arm64 artifacts available from today, but I wanted first to show that a full app can be built with "vanilla" flutter. Kind of chicken and egg problem here.

If you think we should do it the other way, who can implement this for flutter? (build and upload engine)

Reply:
> As mentioned in the commit message, you need a prebuilt arm64 flutter engine (built by hand) to create an arm64 app.

I'm not sure what message you are referring to; the PR description just says that it adds an option to build for Windows ARM. This doesn't actually do that though; it adds a flag that doesn't work. That's extremely likely to confuse people who will, reasonably, assume that if there's a flag that says to target Windows ARM, that's what it will do.

> Kind of chicken and egg problem here.

Having it locally working end to end is useful, and having branch in your fork to point people to can be helpful in some cases, but there's no chicken-and-egg problem when it comes to actually creating/landing production PRs; this is a layered system, with the tool depending on the engine artifacts, but not the reverse. There's nothing about landing a (non-functional) flag in the tool that would make landing the engine changes easier.

> If you think we should do it the other way, who can implement this for flutter? (build and upload engine)

What's the current state of engine support for ARM? If you can cross-compile the Windows ARM engine from an x64 machine, then anyone can make the PRs to add it to the recipes. If not, then nobody can do it until/unless we have Windows ARM infra. (This is discussed in the issue referenced in your PR description.)

Reply:
https://github.com/flutter/flutter/issues/62597#issuecomment-1139125987

In this specific comment, you mentioned having a native arm64 build would be enough. Did things changed since?

Reply:
> I'm a bit disappointed, as this PR was opened two weeks ago, mainly to get those kind of comments "Hey, we won't do it this way, thanks".

I'm sorry I wasn't able to review earlier; I have a very deep review queue at the moment.

However, what I'm saying here isn't any different than what I said in the issue months ago. I clearly said in the issue that having engine artifacts was a pre-requisite for tool changes.

> I agree this flag does not work, except if you build your own engine (as stated in the _commit_ message).

Oh, generally nobody is looking at the commit messages for individual commits in a PR. Important information needs to be in the PR description.

> So from there, how can we advance concretely on this? Cross compile Windows ARM engine is the first step? [...] What will be needed in more?

@cbraken laid out a pretty detailed list in the issue: https://github.com/flutter/flutter/issues/62597#issuecomment-1139141621

> If that works, are you willing to help on making it available as an official artifact?

I'm no longer actively working on desktop, so would not be involved in setting up artifact recipes.

> [#62597 (comment)](https://github.com/flutter/flutter/issues/62597#issuecomment-1139125987)
> 
> In this specific comment, you mentioned having a native arm64 build would be enough. Did things changed since?

Nothing has changed since, that's just not what I said. Bolding the important missing part:

>> The first goal though would be to get the engine cross-compiling (x64->arm64), or compiling natively on arm64, or both. (**And if only the later, a CI solution for creating engine artifacts**).

Having only a native arm64 build without also having ARM CI doesn't address the need for engine artifacts.

Reply:
No problem, I understand you were busy.

I'll continue discussion on original issue #62597.